source/s3_auto_ranged_put.c

```diff
@@ -345,9 +347,20 @@ static int s_s3_auto_ranged_put_next_request(
                     0,
                     AWS_S3_REQUEST_DESC_RECORD_RESPONSE_HEADERS);
 
-                aws_byte_buf_init(&request->request_body, meta_request->allocator, meta_request->part_size);
                 request->part_number = auto_ranged_put->threaded_next_request_data.next_part_number;
 
+                size_t request_body_size = meta_request->part_size;
+
+                if (request->part_number == auto_ranged_put->synced_data.total_num_parts) {
```

Comment:
just double checking: part_number starts at 1 and not 0, right?

Reply:
Correct!  S3 part numbers start at 1.